core-blocks/rss/index.php

```
@@ -0,0 +1,159 @@
+		if ( is_admin() || current_user_can( 'manage_options' ) ) {
+			return '<div class="components-placeholder"><div class="notice notice-error"><strong>' . __( 'RSS Error:', 'gutenberg' ) . '</strong> ' . $rss->get_error_message() . '</div></div>';
+		}
+		return;
+	}
+
+	if ( ! $rss->get_item_quantity() ) {
+		$rss->__destruct();
+		unset( $rss );
+		return '<div class="components-placeholder"><div class="notice notice-error">' . __( 'An error has occurred, which probably means the feed is down. Try again later.', 'gutenberg' ) . '</div></div>';
+	}
+
+	$items = (int) $attributes['postsToShow'];
+	if ( $items < 1 || 10 < $items ) {
+		$items = 5;
+	}
+
+	$list_items = '';
+	foreach ( $rss->get_items( 0, $items ) as $item ) {
+		$link = $item->get_link();
+
+		while ( stristr( $link, 'http' ) != $link ) {
+			$link = substr( $link, 1 );
+		}
+
+		$title = esc_html( trim( strip_tags( $item->get_title() ) ) );
+		if ( empty( $title ) ) {
+			$title = __( '(Untitled)', 'gutenberg' );
+		}
+
+		$link = esc_url( strip_tags( $link ) );
+		if ( $link ) {
+			$title = "<a href='$link'>$title</a>";
+		}
+
+		$title = "<div class='wp-block-rss__item-title'>$title</div>";
+
+		$date = '';
+		if ( $attributes['displayDate'] ) {
+			$date = $item->get_date( 'U' );
+
+			if ( $date ) {
+				$date = sprintf(
+					'<time datetime="%1$s" class="wp-block-rss__item-post-date">%2$s</time>',
+					date_i18n( get_option( 'c' ), $date ),
+					date_i18n( get_option( 'date_format' ), $date )
+				);
+			}
+		}
+
+		$author = '';
+		if ( $attributes['displayAuthor'] ) {
+			$author = $item->get_author();
+			if ( is_object( $author ) ) {
+				$author = $author->get_name();
+				$author = '<span class="wp-block-rss__item-author">' . __( 'By', 'gutenberg' ) . ' ' . esc_html( strip_tags( $author ) ) . '</span>';
+			}
+		}
+
+		$excerpt = '';
+		if ( $attributes['displayExcerpt'] ) {
+			$excerpt = @html_entity_decode( $item->get_content(), ENT_QUOTES, get_option( 'blog_charset' ) );
```

phpcs: `Silencing errors is discouraged`